### PR TITLE
Stabilize iterator_repeat_with

### LINKED ISSUE #48169: Tracking issue for std::iter::repeat_with
This is the tracking issue for `std::iter::repeat_with`.
Currently in nightly, PR = #48156.

### PR DESCRIPTION
**Pull request by tmccombs**
Fixes #48169

### CODE REVIEW

File: src/libcore/iter/sources.rs [hunk 1]
```diff
@@ -209,7 +199,7 @@ unsafe impl<A, F: FnMut() -> A> TrustedLen for RepeatWith<F> {}
 /// assert_eq!(None, pow2.next());
 /// ```
 #[inline]
-#[unstable(feature = "iterator_repeat_with", issue = "48169")]
+#[stable(feature = "iterator_repeat_with", since = "1.28.0")]
 pub fn repeat_with<A, F: FnMut() -> A>(repeater: F) -> RepeatWith<F> {
```
Comment (Centril) [inline src/libcore/iter/sources.rs]: There is some documentation re. DEI left that needs to be removed:

> An iterator produced by `repeat_with()` is a `DoubleEndedIterator`. It is important to note that reversing `repeat_with(f)` will produce the exact same sequence as the non-reversed iterator. In other words, `repeat_with(f).rev().collect::<Vec<_>>()` is equivalent to `repeat_with(f).collect::<Vec<_>>()`.

I would replace this with a note saying that it isn't DEI and to ask users who need that to come forward and file an issue explaining their use case.

**Review by Centril — CHANGES_REQUESTED** [review]
Thanks for filing this! There's a minor adjustment needed.

File: src/libcore/iter/sources.rs (hunk 1 above)
Comment (tmccombs) [reply src/libcore/iter/sources.rs]: Fixed.

**Review by Centril — APPROVED** [review]

**Pull request merged**